File: network/src/main/java/com/difft/android/websocket/internal/push/exceptions/StaleDevicesException.java


package com.difft.android.websocket.internal.push.exceptions;

import com.difft.android.websocket.api.push.exceptions.NonSuccessfulResponseCodeException;
import com.difft.android.websocket.internal.push.StaleDevices;

public class StaleDevicesException extends NonSuccessfulResponseCodeException {

  private final StaleDevices staleDevices;

  public StaleDevicesException(StaleDevices staleDevices) {
    super(410);
    this.staleDevices = staleDevices;
  }

  public StaleDevices getStaleDevices() {
    return staleDevices;
  }
}

File: network/src/main/java/com/difft/android/websocket/internal/push/exceptions/StaleDevicesException.kt
package com.difft.android.websocket.internal.push.exceptions

import com.difft.android.websocket.api.push.exceptions.NonSuccessfulResponseCodeException
import com.difft.android.websocket.internal.push.StaleDevices

class StaleDevicesException(
    val staleDevices: StaleDevices
) : NonSuccessfulResponseCodeException(410)
